test/e2e/prometheus_test.go

```diff
@@ -693,6 +693,15 @@ func TestPrometheusOnlyUpdatedOnRelevantChanges(t *testing.T) {
 	name := "test"
 	prometheus := framework.MakeBasicPrometheus(ns, name, name, 1)
 
+	// Adding an annotation to Prometheus lead to high CPU usage in the past
+	// updating the Prometheus StatefulSet in a loop (See
+	// https://github.com/coreos/prometheus-operator/issues/1659). Added here to
+	// prevent a regression.
+	if prometheus.Annotations == nil {
```

Comment:
rather doing this here, I would advocate to do this in `framework.MakeBasicPrometheus(...)` so the caller does not need to worry about initialization.